### PR TITLE
updated log rotate to be able to forward logs

### PR DESCRIPTION
**Pull request by jalogisch**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

5
---
If you use rsyslog - at least on Jessie PI to forward the logfile to some central location rsyslog does not notice the updated logfile. 

After I had added the restart to the post rotate this is working with no issue. As I did not test if this is working on other type of installation that might only be save for a pi based installation.

### CODE REVIEW

File: advanced/logrotate [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
+    # The postrotate script will only be run once (after the old
+    #   logs  have  been  compressed), not once for each log which is rotated.
+    sharedscripts
+
+    # The  lines  between 'postrotate' and 'endscript' are executed after the log is rotated
+    postrotate
+        # If dnsmasq is running, kill it using the user defined signal
+        #[ ! -f /var/run/dnsmasq/dnsmasq.pid ] || kill -USR2 cat /var/run/dnsmasq/dnsmasq.pid
+    	# use pihole restart function
+	/usr/local/bin/pihole restartdns
```
Comment (DL6ER) [inline advanced/logrotate]: Restarting `dnsmasq` should not be done. The current method works w/o problems for basically all users.

**Comment by DL6ER** [comment]
You could also look at this post here which is weakly related:
https://discourse.pi-hole.net/t/moving-the-pi-hole-log-to-another-location-device/2041

File: advanced/logrotate (hunk 1 above)
Comment (jalogisch) [inline advanced/logrotate]: Maybe I'm just holding it wrong:

Read the log file with rsyslog:

```
module(load="imfile" PollingInterval="10")

input(type="imfile"
      File="/var/log/pihole.log"
      StateFile="/var/run/pihole.log.state"
      Tag="pihole"
      Severity="info"
      Facility="local7")
```

And then notice that this file is not picked up after the `echo "" > `.

Maybe it is just my environments where I'm responsible for, but the restart is the only way to have syslog recognize the new added entries.

**Comment by jalogisch** [comment]
thank you @DL6ER for pointing to that post. 

But I just want to forward that log to my central log environment and not move it to another disk/place on the device.

For your [references (lang=de)](https://jalogisch.de/2017/der-eigene-dns-resolver-zuhause/) why I want this.

**Comment by DL6ER** [comment]
> And then notice that this file is not picked up after the echo "" >.
> Maybe it is just my environments where I'm responsible for, but the restart is the only way to have syslog recognize the new added entries.

So the problem is that `rsyslog` is not behaving nice when the log file is flushed. How about restarting this service?

> For your references (lang=de) why I want this.

German is not an issue in our team as we have a native speaker (which is me :smiley: ) on board. We thought about implementing similar statistics like you show with graylog, but we found severe licensing barriers which the others project simply seems to ignore. However, we want to stay out of curt and hence refrained from implementing the nice and fancy IP maps, etc.

**Comment by jalogisch** [comment]
> So the problem is that rsyslog is not behaving nice when the log file is flushed. How about restarting this service?

valid point - will test some modified script before I modify the PR.

**Comment by DL6ER** [comment]
Any progress on this?

**Comment by jalogisch** [comment]
currently I was not able to find a working solution other than restart the DNS. 

I'm in doubt that his can be fixed with the given version on the default raspdian syslog.

**Comment by dschaper** [comment]
Thank you for the submission, I'm closing this for now, but if you find a solution please reopen the PR.

**Pull request closed without merging**